agent/engine/docker_task_engine_test.go

```diff
@@ -527,11 +526,17 @@ func TestSteadyStatePoll(t *testing.T) {
 		// the engine *may* call StopContainer even though it's already stopped
 		client.EXPECT().StopContainer("containerId", stopContainerTimeout).AnyTimes(),
 	)
+	wait.Wait()
+
+	cleanupChan := make(chan time.Time)
+	testTime.EXPECT().After(gomock.Any()).Return(cleanupChan).AnyTimes()
+	client.EXPECT().RemoveContainer(gomock.Any(), gomock.Any()).Return(nil)
+	imageManager.EXPECT().RemoveContainerReferenceFromImageState(gomock.Any()).Return(nil)
```

Comment:
Can we be more deterministic here wrt the parameters of `RemoveContainerReferenceFromImageState `? May be with the the image name?